vendor/bat-native-ads/src/bat/ads/internal/ad_conversions.cc

```diff
@@ -69,7 +90,117 @@ void AdConversions::ProcessQueue() {
   StartTimer(ad_conversion);
 }
 
-void AdConversions::AddToQueue(
+bool AdConversions::OnTimer(
+    const uint32_t timer_id) {
+  if (timer_id != timer_id_) {
+    return false;
+  }
+
+  timer_id_ = 0;
+
+  DCHECK(!queue_.empty());
+  if (queue_.empty()) {
+    return true;
+  }
+
+  AdConversionQueueItemInfo ad_conversion = queue_.front();
+  ProcessQueueItem(ad_conversion);
+
+  return true;
+}
+
+///////////////////////////////////////////////////////////////////////////////
+
+void AdConversions::OnGetAdConversions(
+    const std::string& url,
+    const Result result,
+    const AdConversionList& ad_conversions) {
+  if (result != SUCCESS) {
+    BLOG(ERROR) << "Failed to check ad conversions";
+    return;
+  }
+
+  std::deque<AdHistory> ads_history = client_->GetAdsShownHistory();
+  ads_history = FilterAdsHistory(ads_history);
+  ads_history = SortAdsHistory(ads_history);
+
+  AdConversionList new_ad_conversions = ad_conversions;
+  new_ad_conversions = FilterAdConversions(url, new_ad_conversions);
```

Comment:
would it be worthwhile returning on `new_ad_conversions.empty()`? Might be marginal..

Reply:
Very marginal, so not sure worth it at this time. Thanks